File: backend/app/schemas/teachers.py
"""Teacher schemas — read-only (data synced from Payment)."""

from datetime import date

from pydantic import BaseModel, ConfigDict

from .base import PaginatedList


class TeacherRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    document_id: str
    first_name: str
    last_name: str
    middle_name: str | None = None
    full_name: str = ""  # from model property via from_attributes
    birth_date: date | None = None
    gender: str | None = None
    phone_number: str | None = None
    photo_url: str | None = None
    is_active: bool
    subjects: list | None = None
    teaching_grade_ids: list[int] | None = None
    class_teacher_grade_id: int | None = None


TeacherList = PaginatedList[TeacherRead]

File: frontend/src/lib/api/teachers.ts
import { z } from "zod"

import { api, validated } from "./client"

export const teacherReadSchema = z.object({
  id: z.number(),
  document_id: z.string(),
  first_name: z.string(),
  last_name: z.string(),
  middle_name: z.string().nullable(),
  full_name: z.string(),
  birth_date: z.string().nullable(),
  gender: z.string().nullable(),
  phone_number: z.string().nullable(),
  photo_url: z.string().nullable(),
  is_active: z.boolean(),
  subjects: z.array(z.string()).nullable(),
  teaching_grade_ids: z.array(z.number()).nullable(),
  class_teacher_grade_id: z.number().nullable(),
})
export type TeacherRead = z.infer<typeof teacherReadSchema>

export const teacherListSchema = z.object({
  data: z.array(teacherReadSchema),
  count: z.number(),
})
export type TeacherList = z.infer<typeof teacherListSchema>

export const teachersApi = {
  list: (params: { skip?: number; limit?: number; search?: string } = {}) =>
    api
      .get<unknown>("/api/v1/teachers/", { params })
      .then(validated<TeacherList>(teacherListSchema)),
}
